src/app/parcels/fetchParcelTableData.ts

```diff
@@ -30,52 +35,158 @@ export const getCongestionChargeDetailsForParcels = async (
+            query = filter.methodConfig.method(query, filter.state);
+        }
+    });
+
+    if (sortState.sort && sortState.column.sortMethodConfig?.methodType === PaginationType.Server) {
+        query = sortState.column.sortMethodConfig.method(query, sortState.sortDirection);
+    } else {
+        query = query.order("packing_datetime", { ascending: false });
+    }
+
+    if (typeof start === "number" && typeof end === "number") {
+        query = query.range(start, end);
+    }
+
+    if (parcelIds) {
+        query = query.in("parcel_id", parcelIds);
+    }
+
+    const { data, error } = await query;
 
     if (error) {
+        console.error(error);
         throw new DatabaseError("fetch", "parcel table data");
     }
+    return data ?? [];
+};
+
+export const getParcelsData = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any[]>[],
+    sortState: SortState<ParcelsTableRow>,
+    start?: number,
+    end?: number,
+    parcelIds?: string[]
+): Promise<ParcelsTableRow[]> => {
+    const processingData = await getParcelProcessingData(
+        supabase,
+        filters,
+        sortState,
+        start,
+        end,
+        parcelIds
+    );
+    const congestionCharge = await getCongestionChargeDetailsForParcels(processingData, supabase);
+    const formattedData = processingDataToParcelsTableData(processingData, congestionCharge);
+
+    return formattedData;
+};
+
+export const getParcelsCount = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any>[]
+): Promise<number> => {
+    let query = supabase.from("parcels_plus").select("*", { count: "exact", head: true });
+
+    filters.forEach((filter) => {
+        if (filter.methodConfig.methodType === PaginationType.Server) {
+            query = filter.methodConfig.method(query, filter.state);
+        }
+    });
+
+    const { count, error } = await query;
+    if (error || count === null) {
+        console.error(error);
+        throw new DatabaseError("fetch", "parcels");
+    }
+    return count;
+};
+
+export const getParcelIds = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any>[],
+    sortState: SortState<ParcelsTableRow>
+): Promise<string[]> => {
+    let query = supabase.from("parcels_plus").select("*");
+    filters.forEach((filter) => {
+        if (filter.methodConfig.methodType === PaginationType.Server) {
+            query = filter.methodConfig.method(query, filter.state);
+        }
+    });
+    if (sortState.sort && sortState.column.sortMethodConfig?.methodType === PaginationType.Server) {
+        query = sortState.column.sortMethodConfig.method(query, sortState.sortDirection);
+    } else {
+        query = query.order("packing_datetime", { ascending: false });
+    }
+    const { data, error } = await query;
+    if (error) {
+        console.error(error);
+        throw new DatabaseError("fetch", "parcels");
+    }
+    return data.map((parcel) => parcel.parcel_id ?? "") ?? [];
```

Comment:
this function returns parcel ids for all the parcels, after filters are applied, ie, no pagination. This is used when we toggle the all checkbox to true, in order to populate checkedParcelIds with all the parcel ids, after filters are applied.

Reply:
Should we additionally filter the list so it doens't contain empty strings
```suggestion
    return data.map((parcel) => parcel.parcel_id ?? "").filter(parcelId => parcelId) ?? [];
```
or even
```suggestion
    return data.map((parcel) => parcel.parcel_id).filter(parcelId => parcelId) ?? [];
```

Reply:
what about
`return data.reduce((reducedData, parcel) => {parcel.parcel_id && reducedData.push(parcel.parcel_id);
 return reducedData})`

Reply:
`return data.reduce((reducedData, parcel) => {parcel.parcel_id && reducedData.push(parcel.parcel_id); ` looks good!